src/main/java/com/cloudera/sparkts/api/java/DateTimeIndexFactory.java

```diff
@@ -110,6 +110,14 @@ public static IrregularDateTimeIndex irregular(long[] dts, ZoneId zone) {
     }
 
     /**
+     * Creates a HybridDateTimeIndex composed of the given indices using the provided date-time zone.
+     * It is the responsibility of the caller to ensure that all indices have the same ZoneId as zone.
+     */
+    public static HybridDateTimeIndex hybrid(DateTimeIndex[] indices, ZoneId zone) {
```

Comment:
I was thinking that requiring the caller to provide a zone might make it more obligatory for the caller to get concerned by the fact that he/she should care for the zone.

As for making the checks, I thought not to add this overhead in the method putting in mind internal API calls. However, thinking again, this method should be more convenient than a direct call to the constructor ```new HybridDateTimeIndex(indices, zone)``` Hence, I think I should go for making the checks inside the method.